modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -61,12 +72,32 @@
     public Node getNode();
 
     /**
-     * Called by a Skinnable when the Skin is replaced on the Skinnable. This method
-     * allows a Skin to implement any logic necessary to clean up itself after
-     * the Skin is no longer needed. It may be used to release native resources.
+     * Called once when {@link Skin} is set, after the
+     * previous skin, if any, has been uninstalled via its {@link #dispose()} method.
+     * This method allows a Skin to register listeners, add child nodes, set
+     * required properties and/or event handlers.
```

Comment:
I don't see how it is relevant for this interface to know what happens to the previous skin.  The interface should simply describe what is expected, and not go into detail here.

```suggestion
     * Called when the {@link Skin} is accepted for its control. The skin can now
     * safely make changes to its associated control, like registering listeners, adding 
     * child nodes and modifying properties and event handlers.
     *
     * <p>When the control calls this method, it guarantees a future call to {@link #dispose} 
     * when the skin is removed.
```

Reply:
I do think it's important, but I like John's wording of the second sentence.
I'd rather not add the sentence about future call - there is no such guarantee.